x-pack/plugins/alerts/server/invalidate_pending_api_keys/task.ts

```diff
@@ -0,0 +1,157 @@
+  config: Promise<AlertsConfig>,
+  taskManager?: TaskManagerStartContract
+) {
+  if (taskManager) {
+    scheduleTasks(logger, taskManager, config);
+  }
+}
+
+function registerAlertsInvalidateApiKeysTask(
+  logger: Logger,
+  internalSavedObjectsRepository: Promise<{
+    createInternalRepository: () => ISavedObjectsRepository;
+  }>,
+  taskManager: TaskManagerSetupContract,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Invalidate Alerts API Keys',
+      createTaskRunner: taskRunner(logger, internalSavedObjectsRepository, securityPluginSetup),
+    },
+  });
+}
+
+async function scheduleTasks(
+  logger: Logger,
+  taskManager: TaskManagerStartContract,
+  config: Promise<AlertsConfig>
+) {
+  const interval = (await config).invalidateApiKeysTask.interval;
+  try {
+    await taskManager.ensureScheduled({
+      id: TASK_ID,
+      taskType: TASK_TYPE,
+      schedule: {
+        interval,
+      },
+      state: {},
+      params: {},
+    });
+  } catch (e) {
+    logger.debug(`Error scheduling task, received ${e.message}`);
+  }
+}
+
+function taskRunner(
+  logger: Logger,
+  internalSavedObjectsRepository: Promise<{
+    createInternalRepository: () => ISavedObjectsRepository;
+  }>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  return ({ taskInstance }: RunContext) => {
+    const { state } = taskInstance;
+    return {
+      async run() {
+        let totalInvalidated = 0;
+        try {
+          const repository = (await internalSavedObjectsRepository).createInternalRepository();
+          const apiKeysToInvalidate = await repository.find<InvalidatePendingApiKey>({
+            type: 'invalidate_pending_api_key',
+          });
+          apiKeysToInvalidate.saved_objects.forEach(async (obj) => {
+            const response = await invalidateAPIKey(
+              { id: obj.attributes.apiKeyId },
+              securityPluginSetup
+            );
+            if (response.apiKeysEnabled === true && response.result.error_count > 0) {
+              logger.error(`Failed to invalidate API Key [id="${obj.attributes.apiKeyId}"]`);
+            } else {
+              try {
+                await repository.delete('invalidate_pending_api_key', obj.id);
+                totalInvalidated++;
+              } catch (err) {
+                logger.error(
+                  `Failed to cleanup api key "${obj.attributes.apiKeyId}". Error: ${err.message}`
+                );
+              }
+            }
+          });
+          return {
+            state: {
+              runs: (state.runs || 0) + 1,
+              total_invalidated: totalInvalidated,
+            },
+          };
+        } catch (errMsg) {
+          logger.warn(`Error executing alerting health check task: ${errMsg}`);
+          return {
+            state: {
+              runs: (state.runs || 0) + 1,
+              total_invalidated: totalInvalidated,
+            },
+          };
```

Comment:
I wonder if we should return the latest schedule in both of these scenarios:
```
schedule: {
  interval,
}
```

This could be useful if ever we decide to increase / decrease the value in a future release because it would "migrate" the customer's task if it already existed. (though this change could be done alter as well..)